### PR TITLE
[14.0][IMP] delivery_schenker : submit booking

### PR DESCRIPTION
**Pull request by vnikolayev1**
- Added field for schenker delivery carrier, where user can pick if they want to submit booking on schenker site. In case where it is turned off - user has to submit booking manually.

### CODE REVIEW

File: delivery_schenker/models/delivery_carrier.py [hunk 1]
```diff
@@ -515,6 +522,11 @@ def _prepare_schenker_shipping(self, picking):
                 "homeDelivery": self.schenker_home_delivery,
                 "ownPickup": self.schenker_own_pickup,
                 "pharmaceuticals": self.schenker_pharmaceuticals,
+                # From the Schenker docs:
+                # Defines if booking shall be submitted. If false,
+                # the booking can be edited
+                # in the frontend and MUST be submitted manually.
+                "submitBooking": picking.carrier_id.is_schenker_submit_booking,
```
Comment (mt-software-de) [inline delivery_schenker/models/delivery_carrier.py]:
```suggestion
                "submitBooking": self.is_schenker_submit_booking,
```

File: delivery_schenker/models/delivery_carrier.py [hunk 2]
```diff
@@ -179,6 +179,13 @@ class DeliveryCarrier(models.Model):
         help="If set, this contact will be sent as invoice address to Schenker."
         "\nIf Address ID is set, it will be part of it instead of the sender",
     )
+    is_schenker_submit_booking = fields.Boolean(
+        string="Submit Booking",
+        default=True,
+        help="Submit Booking On Schenker site."
+        "\nIf set, submits booking on site automatically."
+        "\nIf not set, user has to submit booking manually.",
+    )
```
Comment (mt-software-de) [inline delivery_schenker/models/delivery_carrier.py]: Could you please change the name of the field to schenker_submit_booking
```suggestion
    schenker_submit_booking = fields.Boolean(
        string="Submit Booking",
        default=True,
        help="Submit Booking On Schenker site."
        "\nIf set, submits booking on site automatically."
        "\nIf not set, user has to submit booking manually.",
    )
```

File: delivery_schenker/__manifest__.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 {
     "name": "Delivery Schenker",
     "summary": "Delivery Carrier implementation for DB Schenker API",
-    "version": "14.0.1.2.0",
+    "version": "14.0.1.2.1",
```
Comment (mt-software-de) [inline delivery_schenker/__manifest__.py]: Please remove this change. This should be automatically done by the merge bot.

File: delivery_schenker/tests/common.py [hunk 4]
```diff
@@ -58,6 +58,12 @@ def setUpClass(cls):
             }
         )
         cls.sale, cls.picking = cls._create_sale_order(cls)
+        cls.picking.write(
+            {
+                "shipping_weight": 1,
+                "number_of_packages": 1,
+            }
+        )
```
Comment (mt-software-de) [inline delivery_schenker/tests/common.py]: What is the purpose of this change?

**Review by mt-software-de — CHANGES_REQUESTED** [review]
Thanks for contributing

**Review by jbaudoux — APPROVED** [review]
LGTM, not tested

**Comment by jbaudoux** [comment]
@vnikolayev1 Can you rebase and repush as pre-commit failed

**Comment by vnikolayev1** [comment]
@jbaudoux sorry for delay, should be good now

**Comment by mt-software-de** [comment]
> @jbaudoux sorry for delay, should be good now

Please rebase and not merge. To get rid of the second commit.

**Review by mt-software-de — APPROVED** [review]

**Pull request merged**